src/js/actions/menu.js

```diff
@@ -267,6 +267,160 @@ define(function (require, exports) {
     };
 
     /**
+     * Run layer panel performance tests.   
+     *
+     * @private
+     * @return {Promise}
+     */
+    var layerPanelPerformanceTest = function () {
+        var flux = this.flux,
+            applicationStore = flux.store("application"),
+            document = applicationStore.getCurrentDocument(),
+            openDocuments = applicationStore.getOpenDocuments();
+
+        if (openDocuments.size !== 1 || !document.name.match(/vermilion/i)) {
+            window.alert(
+                "To run the performance test, the current document must be " +
+                "the Vermilion file, and there should be only one open document");
+            return Promise.resolve();
+        }
+
+        var continueTest = window.confirm("Please start the Timeline recording, and then hit OK to begin the test.");
+
+        if (!continueTest) {
+            return Promise.resolve();
+        }
+
+        // Mute the other time stamps to make the timeline cleaner.
+        var timeStamp = log.timeStamp;
+        log.timeStamp = _.noop;
+
+        var layerFaceElements,
+            artboardElement,
+            artboardIconElement,
+            artboardVisibilityElement,
+            delayBetweenTest = 1500;
+
+        return Promise
+            .bind(this)
+            .then(function () {
+                // Expand all artboards
+                document.layers.roots.forEach(function (root) {
```

Comment:
It would be better to use promises for this than a `delay` at the end. That is, this should be mapped into an array of promises, and then `Promise.all` should be used to build a single promise that resolves when all the child promises resolve, and then that should be `.then`d with the promise from `deselectAll` below, etc.